standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/cache/CachedStore.java

```diff
@@ -468,6 +509,125 @@ public void run() {
       }
     }
 
+    @VisibleForTesting
+    public static long updateUsingNotificationEvents(RawStore rawStore, long lastEventId) throws Exception {
+      LOG.debug("updating cache using notification events starting from event id " + lastEventId);
+      NotificationEventRequest rqst = new NotificationEventRequest(lastEventId);
+
+      //Add the events which are not related to metadata update
+      rqst.addToEventTypeSkipList(MessageBuilder.INSERT_EVENT);
+      rqst.addToEventTypeSkipList(MessageBuilder.OPEN_TXN_EVENT);
+      rqst.addToEventTypeSkipList(MessageBuilder.COMMIT_TXN_EVENT);
+      rqst.addToEventTypeSkipList(MessageBuilder.ABORT_TXN_EVENT);
+      rqst.addToEventTypeSkipList(MessageBuilder.ALLOC_WRITE_ID_EVENT);
+      rqst.addToEventTypeSkipList(MessageBuilder.ACID_WRITE_EVENT);
+      rqst.addToEventTypeSkipList(MessageBuilder.CREATE_FUNCTION_EVENT);
+      rqst.addToEventTypeSkipList(MessageBuilder.DROP_FUNCTION_EVENT);
+      rqst.addToEventTypeSkipList(MessageBuilder.ADD_PRIMARYKEY_EVENT);
+      rqst.addToEventTypeSkipList(MessageBuilder.ADD_FOREIGNKEY_EVENT);
+      rqst.addToEventTypeSkipList(MessageBuilder.ADD_UNIQUECONSTRAINT_EVENT);
+      rqst.addToEventTypeSkipList(MessageBuilder.ADD_NOTNULLCONSTRAINT_EVENT);
+      rqst.addToEventTypeSkipList(MessageBuilder.DROP_CONSTRAINT_EVENT);
+      rqst.addToEventTypeSkipList(MessageBuilder.CREATE_ISCHEMA_EVENT);
+      rqst.addToEventTypeSkipList(MessageBuilder.ALTER_ISCHEMA_EVENT);
+      rqst.addToEventTypeSkipList(MessageBuilder.DROP_ISCHEMA_EVENT);
+      rqst.addToEventTypeSkipList(MessageBuilder.ADD_SCHEMA_VERSION_EVENT);
+      rqst.addToEventTypeSkipList(MessageBuilder.ALTER_SCHEMA_VERSION_EVENT);
+      rqst.addToEventTypeSkipList(MessageBuilder.DROP_SCHEMA_VERSION_EVENT);
+
+      NotificationEventResponse resp = rawStore.getNextNotification(rqst);
+      List<NotificationEvent> eventList = resp.getEvents();
+      if (eventList == null) {
+        LOG.info("no events to process");
+        return lastEventId;
+      }
+
+      LOG.debug("num events to process" + eventList.size());
+
+      MessageDeserializer deserializer = MessageFactory.getDefaultInstance(rawStore.getConf()).getDeserializer();
```

Comment:
deserializer better to be class member variable

Reply:
deserializer is per event as it may change based on message format